PC/pyconfig.h.in

```diff
@@ -753,4 +753,8 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
 /* Define if libssl has X509_VERIFY_PARAM_set1_host and related function */
 #define HAVE_X509_VERIFY_PARAM_SET1_HOST 1
 
+// Truncate the thread name to 64 characters. The OS limit is 32766 wide
+// characters, but long names aren't of practical use.
+#define PYTHREAD_NAME_MAXLEN 64
```

Comment:
There's no reason for us to artificially limit this at all, but I'd rather be much closer to the real limit than this. "Aren't of practical use" is a value judgement, not a technical limit.

Also, there's no C API for this, so we probably don't need a public C constant for the limit.

Reply:
I changed the limit to  32766 characters.

Reply:
> Also, there's no C API for this, so we probably don't need a public C constant for the limit.

I can rename the macro use `_Py` prefix. But that unrelated to the Windows implementation, so I would prefer to do it in a separated change. Non-Windows platforms use the same macro.

Reply:
> Also, there's no C API for this, so we probably don't need a public C constant for the limit.

I created https://github.com/python/cpython/pull/128945 to make the macro private.